woocommerce/woocommerce-ios
# [Manual taxes M3] Store selected tax rate

Closes: #10669 


## Description

With this PR we implement storing the selected tax rate (UI + Business Logic) so it can be applied to the orders. As we add another component to the NewTaxSelectorView, we take advantage of this PR to rename some elements.

## Testing instructions

There's not an easy way yet to test that the selected tax rate id was properly saved, but we can add a test log to retrieve that stored id. For instance, in EditableOrderViewModel.addTaxRateAddressToOrder [here](https://github.com/woocommerce/woocommerce-ios/blob/139e0c7b0b0f5b70739f936c9bb269719eb6bdaf/WooCommerce/Classes/ViewRelated/Orders/Order%20Creation/EditableOrderViewModel.swift#L553), you can add:

```
stores.dispatch(AppSettingsAction.loadSelectedTaxRateID(siteID: siteID) { taxRateID in
            debugPrint("stored tax rate id", taxRateID)
        })
```
With a store that has tax rates and the tax setting is based on the customer shipping/billing address:

1. Go to Orders
2. Tap on + to create a new order
3. Tap on Set New Tax Rate
4. Turn the bottom switch on to add the rate to all created orders
5. Select an order
6. See in the Xcode console that the selected tax rate id matches the selected tax rate
7. Go back to the tax selector. Let the switch off. Check that it doesn't store the tax rate id in this case

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/1864060/13224b28-5fd3-457f-b5cc-e70f22e235ec

---
